policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/PolicyApplication.java

```diff
@@ -16,26 +16,133 @@
+            final Collection<RunnableGraph<?>> runners,
+            final Materializer materializer,
+            final PolicyServiceCachingProxy service,
+            final PolicyConfiguration policyConfig,
+            final UserConfiguration userConfig,
+            final ResourceConfiguration resourceConfig,
+            @Qualifier("threadPoolTaskExecutor") final Executor executor) {
+        this.runners = new HashSet<>(runners);
+        this.materializer = materializer;
+        this.service = service;
+        this.policyConfig = policyConfig;
+        this.userConfig = userConfig;
+        this.resourceConfig = resourceConfig;
+        this.executor = executor;
+    }
+
+    /**
+     * Application entrypoint, creates and runs a spring application, passing in the given command-line args
+     *
+     * @param args command-line arguments passed to the application
+     */
     public static void main(final String[] args) {
-        LOGGER.debug("PolicyApplication started with: {} {} {}", PolicyApplication.class.toString(), "main", Arrays.toString(args));
+        LOGGER.debug("PolicyApplication started with: {}", (Object) args);
         new SpringApplicationBuilder(PolicyApplication.class).web(WebApplicationType.SERVLET)
                 .run(args);
     }
 
+    /**
+     * Init post construct.
+     * Pre-population profile used in the KafkaContractTests for pre-populating the cache.
+     * Example profile used for pre-populating the cache for the example
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    @Profile({"pre-population, example"})
+    public void initPostConstruct() {
+        // Add example Policies to the policy-service cache
+        LOGGER.debug("Pre-populating using policy config: {}", policyConfig.getClass());
+        LOGGER.debug("Pre-populating using user config: {}", userConfig.getClass());
+        LOGGER.debug("Pre-populating using resource config: {}", resourceConfig.getClass());
+        policyConfig.getPolicies().stream()
+                .peek(e -> LOGGER.info("pre-pop stream {}", e))
+                .forEach((PolicyPrepopulationFactory factory) -> {
+                    //Build Resource Rules
+                    Entry<Resource, Rules<LeafResource>> resourceMap = factory.buildResourceRules(resourceConfig.getResources());
+                    service.setResourceRules(resourceMap.getKey(), resourceMap.getValue());
+
+                    //Build Record Rules
+                    Entry<Resource, Rules<Serializable>> recordMap = factory.buildRecordRules(resourceConfig.getResources());
+                    service.setRecordRules(recordMap.getKey(), recordMap.getValue());
+                });
+    }
+
+    /**
+     * Runs all available Akka {@link RunnableGraph}s until completion.
+     * The 'main' threads of the application during runtime are the completable futures spawned here.
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    public void serveForever() {
```

Comment:
Shouldn't we have a single EventListener where we do the prepop and *then* do akka stuff? Otherwise slow prepop and quick akka leads to invalid results for a short time-window at start-up

Reply:
I can merge the two, if the question about profiles above is resolved

Reply:
Suggest merge, have a boolean `prepopulateCache` flag or similar if you want the functionality disabled